controllers/sync.go

```diff
@@ -83,29 +66,84 @@ func chooseUpdatingComponents(spec ytv1.YtsaurusSpec, needUpdate []components.Co
 		return nil, "All components are uptodate"
 	}
 
-	switch configuredSelector {
-	case ytv1.UpdateSelectorEverything:
-		if needFullUpdate {
+	if len(configuredSelectors) == 1 && configuredSelectors[0].Class == consts.ComponentClassEverything {
+		if needFullUpdate(needUpdate) {
+			// Here we update not only components that are not up-to-date, but all cluster.
 			return convertToComponent(allComponents), ""
 		} else {
 			return canUpdate, ""
 		}
-	case ytv1.UpdateSelectorMasterOnly:
-		return canUpdate, ""
-	case ytv1.UpdateSelectorTabletNodesOnly:
-		return canUpdate, ""
-	case ytv1.UpdateSelectorDataNodesOnly, ytv1.UpdateSelectorExecNodesOnly, ytv1.UpdateSelectorStatelessOnly:
-		return canUpdate, ""
-	default:
-		return nil, fmt.Sprintf("Unexpected update selector %s", configuredSelector)
 	}
+	return canUpdate, ""
+}
+
+func needFullUpdate(needUpdate []components.Component) bool {
+	statelessSelector := []ytv1.ComponentUpdateSelector{{Class: consts.ComponentClassStateless}}
+	for _, comp := range needUpdate {
+		component := ytv1.Component{
+			Name: comp.GetShortName(),
+			Type: comp.GetType(),
+		}
+		isStateless := canUpdateComponent(statelessSelector, component)
+		if !isStateless && component.Type != consts.DataNodeType {
```

Comment:
Data nodes already considered not stateless in canUpdateComponent, so no need to check